ios/brave-ios/Sources/BraveVPN/Components/BuyVPN/Paywall/BraveVPNPaywallView.swift

```diff
@@ -130,48 +71,96 @@ public struct BraveVPNPaywallView: View {
-        return
-      }
-      
-      if iapObserverManager.isReceiptValidationRequired {
-        Task {
-          _ = await BraveVPN.validateReceiptData()
+    .onChange(of: iapObserverManager.paymentStatus) { status in
+      if case .failure(let error) = status {
+        resetTheRestoreTimerIfNecessary()
+
+        if case .transactionError(let err) = error, err?.code == SKError.paymentCancelled {
+          return
         }
-      }
-      
-      paymentStatus = .success
-      
-      installVPNProfile?()
-    }
-    .onChange(of: iapObserverManager.isPurchaseFailedError) { error in
-      resetTheRestoreTimerIfNecessary()
 
-      paymentStatus = .failure
+        isShowingPurchaseAlert = true
+      } else if case .success(let receiptValidationRequired) = status {
+        resetTheRestoreTimerIfNecessary()
 
-      if case .transactionError(let err) = error, err?.code == SKError.paymentCancelled {
-        return
+        if receiptValidationRequired {
+          Task {
+            do {
+              _ = try await BraveVPN.validateReceiptData()
+            } catch {
+              Logger.module.error("Error validating receipt: \(error)")
+            }
+          }
+        }
+
+        presentationMode.dismiss()
+        installVPNProfile()
+      }
+    }
+    .onReceive(
+      NotificationCenter.default.publisher(for: UIDevice.orientationDidChangeNotification)
+    ) { _ in
+      let new = UIDevice.current.orientation
+      if new != .faceUp && new != .faceDown {
+        orientation = new
       }
-      
-      isShowingPurchaseAlert = true
+    }
+    .onAppear {
+      orientation = UIDevice.current.orientation
     }
     .onDisappear {
       iapRestoreTimer?.cancel()
     }
   }
 
+  private var horizontalContentView: some View {
+    HStack {
+      VStack(alignment: .leading, spacing: 40) {
+        BraveVPNPremiumUpsellView()
+        BraveVPNPoweredBrandView(isFreeTrialAvailable: isFreeTrialAvailable)
+        Spacer()
+      }
+      VStack(spacing: 8.0) {
+        tierSelection
+        BraveVPNSubscriptionActionView(
+          refreshCredentials: {
+            refreshCredential()
+          },
+          redeedPromoCode: {
+            redeemPromoCode()
+          }
+        )
+      }
+    }
+  }
+
+  private var verticalContentView: some View {
+    VStack(spacing: 8.0) {
+      BraveVPNPremiumUpsellView()
+        .padding(.top, 24.0)
+        .padding(.bottom, 8.0)
+      separatorView
+        .padding(.horizontal, -16.0)
+      BraveVPNPoweredBrandView(isFreeTrialAvailable: isFreeTrialAvailable)
+        .padding()
+      tierSelection
+      BraveVPNSubscriptionActionView(
+        refreshCredentials: {
+          refreshCredential()
+        },
+        redeedPromoCode: {
+          redeemPromoCode()
+        }
+      )
+      .padding(.bottom, 8.0)
+    }
+  }
+
+  private var separatorView: some View {
+    Color(braveSystemName: .primitivePrimary25)
+      .frame(height: 1.0)
```

Comment:
May want to use https://developer.apple.com/documentation/swiftui/environmentvalues/pixellength